src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1563,6 +1577,114 @@ def _get_linux_multicontainer_encoded_config_from_file(file_name):
+                                                            value=deployment_storage_conn_string))
+            update_flex_functionapp_configuration(cmd, resource_group_name, name, configs)
+            
+            functionapp_deployment_storage["authentication"]["type"] = deployment_storage_auth_type
+            functionapp_deployment_storage["authentication"]["userAssignedIdentityResourceId"] = None
+            functionapp_deployment_storage["authentication"]["storageAccountConnectionStringName"] = "DEPLOYMENT_STORAGE_CONNECTION_STRING"
+        elif deployment_storage_auth_type == 'SystemAssignedIdentity':
+            assign_identities = ['[system]']
+            functionapp_deployment_storage["authentication"]["type"] = deployment_storage_auth_type
+            functionapp_deployment_storage["authentication"]["userAssignedIdentityResourceId"] = None
+            functionapp_deployment_storage["authentication"]["storageAccountConnectionStringName"] = None
+        elif deployment_storage_auth_type == 'UserAssignedIdentity':
+            assign_identities = [deployment_storage_auth_value]
+            functionapp_deployment_storage["authentication"]["type"] = deployment_storage_auth_type
+            deployment_storage_user_assigned_identity = _get_or_create_user_assigned_identity(cmd, resource_group_name, name, deployment_storage_auth_value, functionapp_location)
+            functionapp_deployment_storage["authentication"]["userAssignedIdentityResourceId"] = deployment_storage_user_assigned_identity.id
+            functionapp_deployment_storage["authentication"]["storageAccountConnectionStringName"] = None
+            assign_identities = [deployment_storage_user_assigned_identity.id]
+        else:
+            raise ValidationError("Invalid value for --deployment-storage-auth-type. Please try again with a valid value.")
+    functionapp["properties"]["functionAppConfig"] = function_app_config
+
+    result = update_flex_functionapp(cmd, resource_group_name, name, functionapp)
+    logger.warning("Updated deployment storage for function app '%s'", result)
+    
+    client = web_client_factory(cmd.cli_ctx)
+    functionapp = client.web_apps.get(resource_group_name, name)
+    if (deployment_storage_auth_type != 'StorageAccountConnectionString' and assign_identities is not None):
+        identity = assign_identity(cmd, resource_group_name, name, assign_identities,'Contributor', None, None)
+        logger.warning("Assigned identity '%s' to function app '%s'", identity, name)
+        functionapp.identity = identity
+
+    if deployment_storage_auth_type == 'SystemAssignedIdentity':
+        _assign_deployment_storage_managed_identity_role(cmd.cli_ctx, deployment_storage, functionapp.identity.principal_id)
+        
+    poller = client.web_apps.begin_create_or_update(resource_group_name, name, functionapp)
+    functionapp = LongRunningOperation(cmd.cli_ctx)(poller)
+    return functionapp
```

Comment:
Let's return specifically the deployment storage config instead of returning the function app (specially since this would not even show the new deployment storage config)

Reply:
Returning the full function object. At the moment, api  is returning an empty object. Will return full object after testing with real backend.